catalog/core/catalog-core-sourcemetricsplugin/src/main/java/ddf/catalog/metrics/source/SourceMetricsImpl.java

```diff
@@ -197,6 +206,15 @@ public QueryResponse process(QueryResponse input)
       for (Map.Entry<String, Integer> source : totalHitsPerSource.entrySet()) {
         updateMetric(source.getKey(), QUERIES_TOTAL_RESULTS_SCOPE, source.getValue());
       }
+
+      // Latency per Source
+      for (Map.Entry<String, Serializable> property : input.getProperties().entrySet()) {
```

Comment:
✏️I would suggest pulling this loop out into a private `updateSourceLatencyMetrics` method to keep the `process` method more focused.